ansible/pbTestScripts/testScript.sh

```diff
@@ -209,88 +210,54 @@ startVMPlaybookWin()
 	# Run the ansible playbook on the VM & logs the output.
 	ansible-playbook -i playbooks/AdoptOpenJDK_Windows_Playbook/hosts.win -u vagrant --skip-tags jenkins,adoptopenjdk,build playbooks/AdoptOpenJDK_Windows_Playbook/main.yml 2>&1 | tee $WORKSPACE/adoptopenjdkPBTests/logFiles/$folderName.$branchName.$OS.log
 	echo The playbook finished at : `date +%T`
-	searchLogFiles $OS
-	if [[ "$testNativeBuild" = true ]]; then
-		echo "Building a JDK"
+	local pbFailed=$(searchLogFiles $OS)
+        if [[ "$testNativeBuild" = true && "$pbFailed" == 0 ]]; then
 		# Runs the build script via ansible, as vagrant powershell gives error messages that ansible doesn't. 
         	# See: https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/942#issuecomment-539946564
         	cd $WORKSPACE/adoptopenjdkPBTests/$folderName-$branchName/ansible
 		ansible all -i playbooks/AdoptOpenJDK_Windows_Playbook/hosts.win -u vagrant -m raw -a "Start-Process powershell.exe -Verb runAs; cd C:/; sh C:/vagrant/pbTestScripts/buildJDKWin.sh"
 		echo The build finished at : `date +%T`
 		if [[ "$runTest" = true ]]; then
-			echo "Running test against the built JDK"
 			# Runs a script on the VM to test the built JDK
 			ansible all -i playbooks/AdoptOpenJDK_Windows_Playbook/hosts.win -u vagrant -m raw -a "sh C:/vagrant/pbTestScripts/testJDKWin.sh"
 			echo The test finished at : `date +%T`
 		fi
 	fi
-        if [[ "$vmHalt" = true ]]; then
-                vagrant halt
-        fi
 }
 
-destroyVM()
+searchLogFiles()
 {
-	echo "Destroying Machine . . ."
-	vagrant destroy -f
-	echo "Removing Work folder"
-	rm -rf $WORKSPACE/adoptopenjdkPBTests/$folderName-$branchName
-	echo "==$WORKSPACE/adoptopenjdkPBTests/=="
-	ls -la $WORKSPACE/adoptopenjdkPBTests
+        local OS=$1
+        cd $WORKSPACE/adoptopenjdkPBTests/logFiles
+        echo
+        if grep -q 'failed=[1-9]\|unreachable=[1-9]' *$folderName.$branchName.$OS.log
+        then
+                return 1;
+        elif grep -q '\[ERROR\]' *$folderName.$branchName.$OS.log
+        then
+                return 1;
+        elif grep -q 'failed=0' *$folderName.$branchName.$OS.log
```

Comment:
If this is the only cast that can return zero (since the else clause returns `1`, why have all the extra checks?